lib/random_team_allocator.ex

```diff
@@ -0,0 +1,46 @@
+defmodule TeamBuilder.RandomTeamAllocator do
+  def assign_teams(members, %{team_type: :fixed, options: fixed_count} = team_type, seed_state) do
+    teams = _assign_teams(members, team_type, seed_state)
+    {teams, :rand.export_seed()}
+  end
+
+  defp _assign_teams([], _, seed_state), do: []
+
+  defp _assign_teams(members, %{team_type: :fixed, options: fixed_count} = team_type, seed_state) do
+    {selection, remainder} = member_selection(members, fixed_count, seed_state)
+    with_team_number(selection) ++ _assign_teams(remainder, team_type, seed_state)
+  end
+
+  defp member_selection(members, team_count, seed) do
+    members
+    |> take_random_pure(team_count, seed)
+    |> partition(members)
+  end
+
+  defp take_random_pure(members, team_count, seed) do
+    :rand.seed(seed)
+    Enum.take_random(members, team_count)
```

Comment:
Now `take_random_pure` is pure in the sense of "will return the same result if the same arguments are passed". That is good :)

But it still is doing a side effect that someone else is depending upon. Specifically, `take_random_pure` is still changing the global seed that `assign_teams` is depending upon with `:rand.export_seed()` in order to be passed at some point again into `take_random_pure`.

In a sense, it is not having the details at the same level of abstraction.
Specifically: `:rand.seed(seed)` and `:rand.export_seed()` are at the same level of abstraction, but they are placed at different levels of abstraction in `TeamBuilder`: `take_random_pure` which is using `:rand.seed(seed)` is very abstract (general use) and `assign_teams` which is using `:rand.export_seed()` is very concrete (specific to this app).

Reference: https://en.wikipedia.org/wiki/Pure_function